src/d3d/nvapi_d3d_low_latency_device.cpp

```diff
@@ -30,25 +70,38 @@ namespace dxvk {
         if (d3dLowLatencyDevice == nullptr)
             return false;
 
-        return SUCCEEDED(d3dLowLatencyDevice->GetLatencyInfo(latencyResults));
+        bool result = SUCCEEDED(d3dLowLatencyDevice->GetLatencyInfo(latencyResults));
+
+        if (result) {
+            for (uint32_t i = 0; i < 64; i++) {
+                uint64_t frameID = latencyResults->frame_reports[i].frameID;
+                latencyResults->frame_reports[i].frameID = GetFrameIdGenerator(device)->GetApplicationFrameId(frameID);
```

Comment:
I believe `GetFrameIdGenerator(device)` can be safely moved outside the loop to lock the map mutex and perform generator lookup only once instead of up to 64 times.